projects/packages/my-jetpack/src/class-initializer.php

```diff
@@ -280,6 +282,23 @@ static function ( $slug ) {
 		return true;
 	}
 
+	/**
+	 * Determines whether the user has come from a host partner we can recognize.
+	 *
+	 * @return string
+	 */
+	public static function is_user_from_host_partner() {
+		require_once __DIR__ . '/utils/nameservers.php';
+		$dns_records = get_nameserver_dns_records();
```

Comment:
Is this worth caching for some period? 
It would be nice not to look this up each time

Reply:
Implemented caching